dhis-2/dhis-services/dhis-service-reporting/src/main/java/org/hisp/dhis/datasetreport/impl/DefaultDataSetReportService.java

```diff
@@ -316,13 +318,56 @@ else if ( formType.isSection() )
             tmpDataSet.getSections().add( section );
         }
 
-        return getSectionDataSetReport( tmpDataSet, period, unit, filters, selectedUnitOnly );
+        return getSectionDataSetReport( tmpDataSet, periods, unit, filters, selectedUnitOnly );
     }
 
     // -------------------------------------------------------------------------
     // Supportive methods
     // -------------------------------------------------------------------------
 
+    private String formatPeriods( List<Period> periods, I18nFormat format )
+    {
+        Collections.sort( periods );
+
+        List<String> formattedPeriods = periods.stream()
+            .map( p -> format.formatPeriod( p ) )
+            .collect( Collectors.toList() );
+
+        return Joiner.on( ", " ).join( formattedPeriods );
+    }
+
+    /**
+     * Returns the sum of values for the list of periods, but returns null if
+     * all the values are null.
+     *
+     * @param dataElement the data element.
+     * @param periods the periods.
+     * @param unit the organisation unit.
+     * @param optionCombo the category option combo.
+     * @return the combined value.
```

Comment:
Personally I find it unnecessary to add javadoc on parameters and return of private methods when you mostly wanted to explain with a comment how it works. Its fine to remove these if there isn't anything more so say.

Reply:
I like it, thanks.